syntax = "proto3";

package exo.networking.grpc;

import "exo/networking/grpc/node_service.proto";

service FileService {
  // Check if a peer has a specific shard
  rpc GetShardStatus (GetShardStatusRequest) returns (GetShardStatusResponse) {}
  
  // Transfer a shard file from one peer to another
  rpc TransferShard (stream ShardChunk) returns (stream TransferStatus) {}
}

message GetShardStatusRequest {
  Shard shard = 1;
  string inference_engine_name = 2;
}

message GetShardStatusResponse {
  bool has_shard = 1;
  optional string local_path = 2;
  optional int64 file_size = 3;
} 